chain/chain/src/store.rs

```diff
@@ -898,6 +906,31 @@ impl ChainStoreAccess for ChainStore {
         read_with_cache(&*self.store, ColTransactions, &mut self.transactions, tx_hash.as_ref())
             .map_err(|e| e.into())
     }
+
+    fn get_key_value_changes(
+        &self,
+        block_hash: &CryptoHash,
+        account_id: &AccountId,
+        key_prefix: &[u8],
+    ) -> Result<StateChanges, Error> {
+        let common_key_prefix_len = block_hash.as_ref().len() + 1 + account_id.len() + 1;
+        let mut storage_key = Vec::with_capacity(common_key_prefix_len + key_prefix.len());
+        storage_key.extend_from_slice(block_hash.as_ref());
+        storage_key.extend_from_slice(b"\0");
+        storage_key.extend_from_slice(account_id.as_bytes());
+        storage_key.extend_from_slice(b",");
+        storage_key.extend_from_slice(key_prefix);
```

Comment:
@nearmax I don't like this ad-hoc "serialization". Is there a dedicated function doing this consistently throughout the code-base?

Reply:
Addressed in another comment.